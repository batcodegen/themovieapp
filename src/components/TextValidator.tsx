import {StyleSheet, Text, View} from 'react-native';
import React from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

interface TextValidatorProps {
  isValid: boolean;
  title: string;
}

const TextValidator = ({isValid, title}: TextValidatorProps) => {
  return (
    <View style={styles.container}>
      <Icon
        name={
          isValid
            ? 'checkbox-marked-circle-outline'
            : 'checkbox-blank-circle-outline'
        }
        size={20}
        color={isValid ? '#1fee3e' : 'gray'}
      />
      <Text style={[styles.validationText, isValid && styles.strikethrough]}>
        {title}
      </Text>
    </View>
  );
};

export default TextValidator;

const styles = StyleSheet.create({
  container: {flexDirection: 'row', alignItems: 'center', width: '70%'},
  validationText: {color: 'white', marginHorizontal: 10},
  strikethrough: {
    textDecorationLine: 'line-through',
    textDecorationColor: 'gray',
    textDecorationStyle: 'solid',
  },
});
