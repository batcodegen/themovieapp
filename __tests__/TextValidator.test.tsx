import React from 'react';
import {render, screen} from '@testing-library/react-native';
import TextValidator from '../src/components/TextValidator';

describe('TextValidator component', () => {
  test('renders Login correctly', async () => {
    render(<TextValidator isValid={true} title="test" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
