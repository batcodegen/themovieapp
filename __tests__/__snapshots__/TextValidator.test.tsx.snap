// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextValidator component renders Login correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "width": "70%",
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1fee3e",
          "fontSize": 20,
        },
        undefined,
        {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ó°„´
  </Text>
  <Text
    style={
      [
        {
          "color": "white",
          "marginHorizontal": 10,
        },
        {
          "textDecorationColor": "gray",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
